src/libexpr/eval-cache.hh

```diff
@@ -56,7 +56,10 @@ struct failed_t {};
 struct int_t { NixInt x; };
 typedef uint64_t AttrId;
 typedef std::pair<AttrId, Symbol> AttrKey;
-typedef std::pair<std::string, NixStringContext> string_t;
+// The order is not supposed to matter, we but we are precisely tracking
+// evaluation
+typedef std::vector<NixStringContextElem> NixStringContextExact;
+typedef std::pair<std::string, NixStringContextExact> string_t;
```

Comment:
I'm not sure whether we really need that. The [definition of the String type](https://github.com/NixOS/nix/blob/d3e2394e9106416e57cd0da10facd8db00e622e6/src/libexpr/value.hh#L188) mentions that the context must be sorted any way. So I think it's OK to just store it in an arbitrary order in the db

Reply:
Oh nice, I'll do what you say and cite that comment. Thanks!